client/src/main/java/io/pravega/client/connection/impl/FlowToBatchSizeTracker.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.connection.impl;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.pravega.common.util.SimpleCache;
+import io.pravega.shared.protocol.netty.AppendBatchSizeTracker;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Duration;
+import java.util.function.BiConsumer;
+
+/**
+ * Class that associates flows with {@link AppendBatchSizeTracker} instances. When connection pooling is disabled, this
+ * class is expected to contain only one flow associated to one {@link AppendBatchSizeTracker}. On the other hand, when
+ * connection pooling is enabled, this class is expected to contain multiple of such associations. The internal flow
+ * to {@link AppendBatchSizeTracker} map is implemented via a {@link SimpleCache} object, which in addition to thread-safe
+ * operations, it also provides an eviction mechanism. This will keep low the memory requirements of this class along time.
+ * Note that if a flow entry is evicted and then the flow is used again, the only consequence will be that its previous
+ * batching information would be lost and the flow will start operating with the batching algorithm defaults.
+ */
+@Slf4j
+public class FlowToBatchSizeTracker {
+
+    /**
+     * Maximum time that the want to keep an inactive flow entry in the cache.
+     */
+    private static final Duration TRACKER_CACHE_EXPIRATION_TIME = Duration.ofSeconds(60);
```

Comment:
This value can be same as `io.pravega.client.connection.impl.TcpClientConnection#SOCKET_TIMEOUT_MS`

Reply:
Done.